src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/mturk/MturkPlatform.java

```diff
@@ -63,8 +63,18 @@ public Boolean isCalibrationAllowed() {
     }
 
     @Override
-    public Optional<WorkerIdentification> getWorker() {
-        return Optional.of(this);
+    public Optional<WorkerIdentification> getWorker(Map<String, String[]> params) throws UnidentifiedWorkerException {
+        String workerId = "";
+        if (params != null) {
+            String[] workerIdArray = params.get("mTurkWorkerId");
+
+            if (workerIdArray != null && workerIdArray.length > 0) {
+                workerId = workerIdArray[0];
+            } else {
+                throw new UnidentifiedWorkerException("mTurkWorkerId was not set!");
+            }
+        }
+        return Optional.of(WorkerIdentification.findByIdentification(getID(),workerId));
```

Comment:
returning an empty string will probably cause a database error or an error in worker operations.
rather throw an exception here.